utils/src/ooinstall/oo_config.py

```diff
@@ -76,30 +67,50 @@ def to_dict(self):
 
     def is_etcd_member(self, all_hosts):
         """ Will this host be a member of a standalone etcd cluster. """
-        if not self.master:
+        if not self.is_master:
             return False
-        masters = [host for host in all_hosts if host.master]
+        masters = [host for host in all_hosts if host.is_master]
         if len(masters) > 1:
             return True
         return False
 
+    def is_master(self):
+        """ Will this host be an openshift master. """
+        return 'master' in self.roles
+
+    def is_node(self):
+        """ Will this host be an openshift node. """
+        return 'node' in self.roles
+
     def is_dedicated_node(self):
         """ Will this host be a dedicated node. (not a master) """
-        return self.node and not self.master
+        return self.is_node and not self.is_master
 
     def is_schedulable_node(self, all_hosts):
         """ Will this host be a node marked as schedulable. """
-        if not self.node:
+        if not self.is_node:
             return False
-        if not self.master:
+        if not self.is_master:
             return True
 
-        masters = [host for host in all_hosts if host.master]
-        nodes = [host for host in all_hosts if host.node]
+        masters = [host for host in all_hosts if host.is_master]
+        nodes = [host for host in all_hosts if host.is_node]
         if len(masters) == len(nodes):
             return True
         return False
 
+    def is_infra_node(self):
+        """ Will this host be an infrastruction node. """
+        return 'infra' in self.roles
+
+    def is_master_lb(self):
+        """ Is this host a load balancer for all masters. """
+        return 'master_lb' in self.roles
+
+    def is_storage(self):
+        """ Will this host be NFS storage for a registry. """
+        return 'storage' in self.roles
+
 
 class OOConfig(object):
```

Comment:
I don't see any conversion from v1 to v2 happening, we'll need to make sure to handle that case.